src/components/ha-related-items.ts

```diff
@@ -74,38 +73,10 @@ export class HaRelatedItems extends LitElement {
     }
   }
 
-  private _relatedEntities = memoizeOne((entityIds: string[]) =>
-    this._toEntities(entityIds)
+  private _toEntities = memoizeOne((entityIds: string[]) =>
+    entityIds.map((entityId) => this.hass.states[entityId])
   );
```

Comment:
This change will defeat the purpose of MemoizeOne here. You can read about MemoizeOne here: https://www.npmjs.com/package/memoize-one#usage

Previously, we had individual MemoizeOne on entities, groups, automations, scripts, ...
So, if the related scripts don't change, memoize will return the cache result fast upon each rendering. 

With these changes, only one MemoizeOne remains and since we fetch related entities, groups, automations, scripts and so on in sequence, it will never be able to return the cache.

Reply:
I would make this 1 memoized function, that returns all the sorted and filtered entities:

```ts
private _getRelated_ = memoizeOne((this._related) => {entity: [], automation: [], ... });
```